src/main/scala/mesosphere/marathon/health/HealthCheckWorkerActor.scala

```diff
@@ -43,35 +43,42 @@ class HealthCheckWorkerActor extends Actor with ActorLogging {
 
   def doCheck(
     app: AppDefinition, task: Task, launched: Task.Launched, check: HealthCheck): Future[Option[HealthResult]] =
-
-    check.hostPort(launched) match {
-      case None => Future.successful {
-        Some(Unhealthy(task.taskId, launched.appVersion, "Missing/invalid port index and no explicit port specified"))
-      }
-      case Some(port) => check.protocol match {
-        case HTTP  => http(app, task, launched, check, port)
-        case TCP   => tcp(app, task, launched, check, port)
-        case HTTPS => https(app, task, launched, check, port)
-        case COMMAND =>
-          Future.failed {
-            val message = s"COMMAND health checks can only be performed " +
-              "by the Mesos executor."
-            log.warning(message)
-            new UnsupportedOperationException(message)
+    task.effectiveIpAddress(app) match {
```

Comment:
This double nesting is not so nice. I would be nicer to kind of chain the stuff without nesting -- e.g. with `Try`.